eng/pre-deploy-tests.yaml

```diff
@@ -0,0 +1,30 @@
+parameters:
+  AzureSubscription: ''
+  PublishProfile: ''
+  stageName: 'Pre_Deploy_Tests'
+
+stages:
+- stage: ${{ parameters.stageName }}
+  displayName: Pre-Deployment Validation
+  pool:
+    name: Hosted VS2017
+  dependsOn:
+  - Build
+  jobs:
+  - job: Validate_Existing_Services
+    pool:
+      name: Hosted VS2017
+    steps:
+    - download: current
+      artifact: ValidationScripts
+    - download: current
+      artifact: ServiceFabricPublishProfiles
+    - download: current
+      artifact: ReleaseUtilities # for Publish profile parsing helper code
+    - task: AzureCLI@2
+      displayName: Validate Service Fabric Applications
+      inputs:
+        azureSubscription: ${{ parameters.AzureSubscription }}
+        scriptType: ps
+        scriptPath: $(Pipeline.Workspace)/ValidationScripts/Pre-DeployCheckServiceFabric.ps1
+        arguments: '-PublishProfile "$(Pipeline.Workspace)/ServiceFabricPublishProfiles/${{ parameters.PublishProfile }}.xml"'
```

Comment:
@adiaaida In the [original file](https://dev.azure.com/dnceng/internal/_git/dotnet-helix-service?path=%2Feng%2Fpre-deploy-tests.yaml&version=GBmain&line=31&lineEnd=32&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) you pass publish profile argument as env variable? It works both ways so are there any benefits to using envvar vs the way I did it?

Reply:
Not that I know of. I think I was just trying to be extra careful.